pyflakes/checker.py

```diff
@@ -1073,6 +1074,12 @@ def addBinding(self, node, value):
                 break
         existing = scope.get(value.name)
 
+        global_scope = self.scopeStack[-1]
+        if (existing and global_scope.get(value.name) == existing and
+                not existing.assigned):
+            # make sure the variable is in the global scope before setting as assigned
+            existing.assigned = True
```

Comment:
pyflakes running on itself is showing this is probably not quite  right -- or at least it's not happy about `nonlocal`